### PR TITLE
Support for loading data from csv files

### PR DESCRIPTION
**Pull request by SimonKonar**
### Description:

Added support for reading data from Csv files in persistence. The goal is to have more supported formats to extract data from. 

### Checklist:

_Make sure you tick all the boxes bellow if they are true or do not apply:_

- [ +] I have performed a self-review of my code
- [ +] My code follows [Google Python Style](https://google.github.io/styleguide/pyguide.html)
- [ +] I have made my code as simple as possible
- [ +] I have added unit tests and the code coverage has not decreased
- [ +] I have updated the documentation in all relevant places

### CODE REVIEW

File: service/blaze_service.py [hunk 1]
```diff
@@ -23,19 +24,26 @@
...
         Must be without a trailing /.
         """
-        self._patient_service = patient_service
-        self._condition_service = condition_service
-        self._sample_service = sample_service
+        self._patient_service = PatientService(repository_factory.create_sample_donor_repository())
+        self._condition_service = ConditionService(repository_factory.create_condition_repository())
+        self._sample_service = SampleService(repository_factory.create_sample_repository())
+        self._sample_collection_repository = repository_factory.create_sample_collection_repository()
+        # self._patient_service = patient_service
```
Comment (RadovanTomik) [inline service/blaze_service.py:42]: remove commented code

**Review by RadovanTomik — CHANGES_REQUESTED** [review]
Please address comments and update the documentation.

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,6 @@
-FROM python:3.11-alpine
+FROM python:3.12-alpine
```
[the remark this answers is not in the document]
Comment (SimonKonar) [reply Dockerfile]: Version is changed back to 3.11

[the remark this answers is not in the document]
Comment (SimonKonar) [reply Dockerfile]: version reverted back to 3.11

File: main.py [hunk 3]
```diff
@@ -1,36 +1,19 @@
...
-                ConditionXMLRepository(records_path=RECORDS_DIR_PATH,
-                                       condition_parsing_map=PARSING_MAP['condition_map'])),
-            sample_service=SampleService(SampleXMLRepository(records_path=RECORDS_DIR_PATH,
-                                                             sample_parsing_map=
-                                                             PARSING_MAP['sample_map'],
-                                                             type_to_collection_map=TYPE_TO_COLLECTION_MAP)),
-            blaze_url=BLAZE_URL,
-            sample_collection_repository=SampleCollectionJSONRepository(SAMPLE_COLLECTIONS_PATH))
+        repository_factory = get_repository_factory()
+        blaze_service = BlazeService(repository_factory=repository_factory, blaze_url=BLAZE_URL)
```
[the remark this answers is not in the document]
Comment (SimonKonar) [reply main.py]: Blaze no longer uses factory as a parameter.

File: main.py [hunk 4]
```diff
@@ -1,36 +1,19 @@
...
-            condition_service=ConditionService(
-                ConditionXMLRepository(records_path=RECORDS_DIR_PATH,
-                                       condition_parsing_map=PARSING_MAP['condition_map'])),
-            sample_service=SampleService(SampleXMLRepository(records_path=RECORDS_DIR_PATH,
-                                                             sample_parsing_map=
-                                                             PARSING_MAP['sample_map'],
-                                                             type_to_collection_map=TYPE_TO_COLLECTION_MAP)),
-            blaze_url=BLAZE_URL,
-            sample_collection_repository=SampleCollectionJSONRepository(SAMPLE_COLLECTIONS_PATH))
+        repository_factory = get_repository_factory()
```
[the remark this answers is not in the document]
Comment (SimonKonar) [reply main.py:18]: I created function in which RECORDS_FILE_TYPE is checked, and corresponding factory is instantiated, or Exception is thrown. I wanted to do it like this, so main.py is clearer, and when new file types will be added, only this function will change.

File: persistence/csv_util.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+
+
+def check_sample_map_format(sample_parsing_map: dict):
+    """Check if sample_details is available in sample parsing map"""
+    if sample_parsing_map.get("sample_details") is None:
+        raise WrongSampleMapException
+
+
+class WrongSampleMapException(Exception):
```
[the remark this answers is not in the document]
Comment (SimonKonar) [reply persistence/csv_util.py]: moved to separate file, also throwing this exception comes with error message as well

File: persistence/factories/factory_util.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    match RECORDS_FILE_TYPE:
+        case "csv":
+            return CSVRepositoryFactory()
+        case "xml":
+            return XMLRepositoryFactory()
+        case _:
+            raise WrongRecordsFileTypeException
+
+
+class WrongRecordsFileTypeException(Exception):
```
[the remark this answers is not in the document]
Comment (SimonKonar) [reply persistence/factories/factory_util.py]: moved to separate file, also throwing this exception comes with error message as well

File: service/blaze_service.py (hunk 1 above)
Comment (SimonKonar) [reply service/blaze_service.py]: comments removed

**Pull request merged**